ledger/src/blockstore_processor.rs

```diff
@@ -294,6 +294,63 @@ fn execute_batches_internal(
     })
 }
 
+// This fn diverts the code-path into two variants. Both must provide exactly the same set of
+// validations. For this reason, this fn is deliberately inserted into the code path to be called
```

Comment:
> Both must provide exactly the same set of validations

I'm aware that `apply_cost_tracker_during_replay` needs to be handled for unified scheduler ;)